### PR TITLE
Pr mavsdk install

### PR DESCRIPTION
**Pull request by AlexKlimaj**

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -183,24 +183,40 @@ else
...
+# Check if we are on 20.04 or 22.04
+if [ "$(lsb_release -cs)" = "focal" ]; then
+	echo "Ubuntu 20.04 detected, building MAVSDK from source"
+	pushd .
+	sudo rm -rf ~/code/MAVSDK
+	git clone https://github.com/mavlink/MAVSDK.git ~/code/MAVSDK
+	cd ~/code/MAVSDK
+	git submodule update --init --recursive
+	cmake -Bbuild/default -DCMAKE_BUILD_TYPE=Release -H.
+	cmake --build build/default -j8
```
Comment (dakejahl) [inline setup.sh]:
```suggestion
	cmake --build build/default -j$(nproc)
```

File: setup.sh [hunk 2]
```diff
@@ -183,24 +183,40 @@ else
...
-    exit 1
-fi
+# Check if we are on 20.04 or 22.04
+if [ "$(lsb_release -cs)" = "focal" ]; then
+	echo "Ubuntu 20.04 detected, building MAVSDK from source"
+	pushd .
+	sudo rm -rf ~/code/MAVSDK
+	git clone https://github.com/mavlink/MAVSDK.git ~/code/MAVSDK
+	cd ~/code/MAVSDK
+	git submodule update --init --recursive
```
Comment (dakejahl) [inline setup.sh]:
```suggestion
	git clone --recurse-submodules --depth=1 --shallow-submodules https://github.com/mavlink/MAVSDK.git ~/code/MAVSDK
	cd ~/code/MAVSDK
```

File: setup.sh [hunk 3]
```diff
@@ -183,24 +183,40 @@ else
...
+	cmake -Bbuild/default -DCMAKE_BUILD_TYPE=Release -H.
+	cmake --build build/default -j8
+	sudo cmake --build build/default --target install
+	popd
```
Comment (dakejahl) [inline setup.sh:197]:
```suggestion
        sudo ldconfig
	popd
```

File: setup.sh [hunk 4]
```diff
@@ -239,6 +255,8 @@ if [ "$INSTALL_LOGLOADER" = "y" ]; then
 
 	make install
 
+	sudo cp build/logloader /usr/local/bin/logloader
```
Comment (dakejahl) [inline setup.sh:258]: Why are you copying the contents of build? The `make install` above already takes care of installing

**Review by dakejahl — CHANGES_REQUESTED** [review]

File: setup.sh (hunk 2 above)
Comment (dakejahl) [reply setup.sh]: the depth=1 and shallow-submodules options speeds up the download a lot, so it doesn't have to pull and index all of the repo history

File: setup.sh (hunk 3 above)
Comment (dakejahl) [reply setup.sh:197]: ensures linker cache is updated

File: setup.sh (hunk 1 above)
Comment (dakejahl) [reply setup.sh]: uses all available cores

File: setup.sh (hunk 4 above)
Comment (AlexKlimaj) [reply setup.sh:258]: Doesn't appear that make install is updating the binary.

Comment (dakejahl) [reply setup.sh:258]: I pushed a logloader commit which performs a clean build on `make install`. Should fix the issue

**Comment by dakejahl** [comment]
Brought in here
https://github.com/ARK-Electronics/ark_companion_scripts/pull/22/commits/51bae37163dac4e511b5fc04ee7cead3045d8674

**Pull request closed without merging**